### DISCUSSION TITLE
'Get.create' memory management

### DISCUSSION THREAD

**Discussion by tak8997** [jonataslaw/getx#2947] [Q&A]
There are many usage examples for Get.create(). In particular, there are examples that are frequently used with ListView and GetWidget, and you may have seen examples of initializing the controller in each Item using Get.find().
However, Get.create() is permanent by default. How should I manage memory? Do you explicitly delete it based on things like life cycle? Or will it be automatically gc if not used?

**Comment by jonataslaw** [comment]
Only the "factory" is permanent (the function that creates instances of that class) and not the instance itself. Furthermore, GetWidget removes the controller from memory as soon as the widget leaves the tree. However, be aware that this API has undergone profound changes in V5, and probably if you use it, you will have to refactor your code.

**Reply by tak8997** [reply]
Thank for reply. but i have a couple of questions :)
1. what's that mean 'only the factory is permanent(the function that creates instances of that class)' in detail?? I know permanent is used in a lot of places. Do all of those permanent mean the same thing?
2. By setting permanent to true in the initialBinding, i'm creating a singleton object. is this the correct way?
